### ISSUE TITLE
How to calculate type of return expression in TypeScript

### ISSUE THREAD

**Issue by unlight** [typescript-eslint/typescript-eslint#960]
It is not an issue, but it is more question about parser.
How I can get type of return for method.
For example:
```ts
class App { 
  initialized() { 
    return obs();
  } 
}
```
`obs` has signature defined outside of current file and it returns `Observable`. So, can definitely say that result of `app.initialized()` is also `Observable`.
How can I retrieve this information?

I was trying to find something useful in `context.parserServices`
```ts
const parserServices = context.parserServices;
const tc = parserServices!.program!.getTypeChecker();
```
But without luck
```ts
                parserOptions: {
                    project: 'tsconfig.json',
                }
```
Error in unit tests
```
 A fatal parsing error occurred: Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: <input>.
The extension for the file () is non-standard. You should add "parserOptions.extraFileExtensions" to your config.
```

**Comment by MaxMilton** [comment]
What's the file extention of the file you're trying to parse? Did you try adding it to `parserOptions.extraFileExtensions` like the error suggested?

**Comment by bradzacher** [comment]
> Error in unit tests

you cannot pass a file contents without a filename.
your unit tests must provide a filename that matches the tsconfig you setup.

We do this as part of our override of the rule tester:
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/tests/RuleTester.ts

We have plenty of rules that use type information, you should browse through our source code for examples of how to consume types.

**Comment by unlight** [comment]
> your unit tests must provide a filename that matches the tsconfig you setup.

Yes, I debugged this...

Second part about `parserServices`
I think I can research this project https://github.com/cartant/eslint-plugin-rxjs/tree/master/tests
It active uses `parserServices` to check `Observable` type.

**Comment by bradzacher** [comment]
i mean sure that's one option, or there's the rules in this repo which check various things about types across codebases.

Going to close this as there's no active bug. feel free to continue commenting.
